packages/strapi/lib/Strapi.js

```diff
@@ -1,106 +1,235 @@
+      await this.freeze();
+      // Launch server.
+      this.server.listen(this.config.port, err => {
+        if (err) {
+          console.log(err);
+        }
+
+        this.log.info('Server started in ' + this.config.appPath);
+        this.log.info('Your server is running at ' + this.config.url);
+        this.log.debug('Time: ' + new Date());
+        this.log.debug('Launched in: ' + (Date.now() - this.config.launchedAt) + ' ms');
+        this.log.debug('Environment: ' + this.config.environment);
+        this.log.debug('Process PID: ' + process.pid);
+        this.log.debug(`Version: ${this.config.info.strapi} (node v${this.config.info.node})`);
+        this.log.info('To shut down your server, press <CTRL> + C at any time');
+
+        if (cb && typeof cb === 'function') {
+          cb();
+        }
+      });
+    } catch (e) {
+      this.log.debug(`Server wasn't able to start properly.`);
+      this.log.error(e);
+      this.stop();
+    }
+  }
 
-    this.reload = require('./reload').apply(this);
+  async enhancer() {
+    const connections = {};
 
-    return _instance;
-  }
+    this.server.on('connection', conn => {
+      const key = conn.remoteAddress + ':' + conn.remotePort;
+      connections[key] = conn;
 
-  // Method to initialize instance
-  initialize(cb) {
-    require('./private/initialize').apply(this, [cb]);
-  }
+      conn.on('close', function() {
+       delete connections[key];
+      });
+    });
 
-  // Method to load instance
-  load(configOverride, cb) {
-    require('./load').apply(this, [configOverride, cb]);
-  }
+    this.server.destroy = cb => {
+      this.server.close(cb);
 
-  // Method to start instance
-  start(configOverride, cb) {
-    require('./start').apply(this, [configOverride, cb]);
+      for (let key in connections) {
+        connections[key].destroy();
+      };
+    };
   }
 
-  // Method to stop instance
   stop() {
-    require('./stop').apply(this);
+    // Destroy server and available connections.
+    this.server.destroy();
+    // Kill process.
+    process.exit(0);
   }
 
-  // Method to expose instance globals
-  exposeGlobals(cb) {
-    require('./private/exposeGlobals').apply(this, [cb]);
+  async load() {
+    // Create AST.
+    await Promise.all([
+      nestedConfigurations.call(this),
+      apis.call(this),
+      middlewares.call(this),
+      hooks.call(this)
+    ]);
+
+
+
+    // Populate AST with configurations.
+    await appConfigurations.call(this);
+
+    // Initialize hooks and middlewares.
+    await Promise.all([
+      initializeMiddlewares.call(this),
+      initializeHooks.call(this)
+    ]);
   }
 
-  // Method to run instance bootstrap
-  runBootstrap(cb) {
-    require('./private/bootstrap').apply(this, [cb]);
+  reload() {
+    const reload = function() {
+      if (cluster.isWorker && process.env.NODE_ENV === 'development' && this.config.environments[this.config.environment].server.reload === true) process.send('message');
```

Comment:
You could use the `this.config.currentEnvironment` variable instead. Also, the configuration key is `autoReload` not `reload`.